### ISSUE TITLE
compiler, when it watch mode, resets the screen colors

### ISSUE THREAD

**Issue by emmanueltouzery** [microsoft/TypeScript#29829]
**TypeScript Version:**  3.4.0-dev.20190208

**Search Terms:** terminal, watcher

**Issue**

I'm using linux, and I have a custom terminal theme (I'm using [powerlevel-9k](https://github.com/bhilburn/powerlevel9k)). To make it look more pleasing, I tuned the terminal colors, but I did that through tput commands in my .zshrc:

```
tput initc 0 172 196 200
tput initc 1 800 210 100
tput initc 2 220 560 220
tput initc 3 800 460 180
tput initc 4 350 530 670
tput initc 5 630 380 470
tput initc 6 470 710 760
tput initc 7 810 810 810
tput initc 8 570 570 570
tput initc 9 1000 280 200
tput initc 10 720 710 0
tput initc 11 1000 780 430
tput initc 12 530 760 1000
tput initc 13 820 820 1000
tput initc 14 440 760 830
tput initc 15 910 910 910
```
I am loosing these customizations if I run typescript's compiler tsc in watch mode (tsc -w) and exit it through ctrl-c. When returning to the shell after that, the colors are the original ones, as if the tput commands were never executed. So it seems `tsc -w` is sending a too drastic reset command to the terminal, which does not only erase `tsc`'s output but can also discard some user's customizations to their terminal.

Notice that the issue occurs with the `xterm` and `kitty` terminals, but NOT with `gnome-terminal`, which seems to be a `gnome-terminal` or `vte` bug (see [kitty bug](https://github.com/kovidgoyal/kitty/issues/1367) and [vte bug](https://gitlab.gnome.org/GNOME/vte/issues/91)).

**Expected behavior:**
Running and exiting `tsc -w` shouldn't discard my color customizations to my terminal.

**Actual behavior:**
Running and exiting `tsc -w` discards my color customizations to my terminal.

Note that when searching if that bug was already reported, I stumbled upon the `--preserveWatchOutput` flag and I can confirm that when this flag is enabled, the issue is NOT present.

**Comment by DanielRosenwasser** [comment]
I've hit this, but it's not only `tsc` that runs into this. I guess we can try to intercept SIGINT and send a color reset code? Seems nasty in the current system though. @Tyriar might also know more here.

**Comment by egmontkob** [comment]
As per the VTE investigation, it's an `ESC c` emitted at _startup_ that reverts the palette.

>  I guess we can try to intercept SIGINT and send a color reset code?

What Emmanual is asking for is _not_ to emit such codes.

**Comment by textshell** [comment]
https://github.com/Microsoft/TypeScript/blob/master/src/compiler/sys.ts#L668
```
                clearScreen: () => {
                    process.stdout.write("\x1Bc");
                },
```
But that code is reset. If should be [Erase Display Complete](https://terminalguide.netlify.com/seq/csi_cj-2/) aka `\x1B[2J`

**Comment by orta** [comment]
I took a quick stab at using `"\x1Bc"` but it had weird behavior, where it wouldn't reset back to up to the top, but clear and continue.

![image](https://user-images.githubusercontent.com/49038/61228276-f3428f00-a6f3-11e9-9519-c59ee331df36.png)

Adding both:

```ts
      clearScreen: () => {
                    process.stdout.write('\x1b[0f');
                    process.stdout.write("\x1B[2J");
                },
```
which I saw in [node-clear](https://github.com/bahamas10/node-clear/blob/master/index.js#L19-L23) would always skip the first line too. 

This is unlikely a quick fix.
